### PR TITLE
feat: Moving configuration and auth logic

### PR DESCRIPTION
**Pull request by mnorbury**
This is being moved from cloudquery/cli to allow it to be shared by both the `plugin-sdk` and `cloudquery/cli` components.

This may not be the best place for this code in the long run, but for now we want to share the code between `cloudquery/cli` and `plugin-sdk` without having `cloudquery/cli` introduce a dependency on the `plugin-sdk`.

This PR also modifies the token issuing logic, to only refresh the ID token if the existing token is close to expiry. This is to minimise the API calls needed during the sync upsert process.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.PHONY: test
+test:
+	go test -tags=assert -race ./...
```
Comment (disq) [inline Makefile:3]: `-tags=assert` is for Arrow tests, we don't need this I think... just fyi :) (still it could be beneficial to keep it...)

Comment (mnorbury) [reply Makefile:3]: Ah, I wasn't quite sure what that was for. Good to know.

Comment (hermanschaaf) [reply Makefile:3]: Yeah good call, let's remove it :)

**Review by hermanschaaf — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (mnorbury) [reply Makefile:3]: Sorry, missed the request - follow up [here](https://github.com/cloudquery/cloudquery-api-go/pull/41)

**Pull request merged**